### PR TITLE
Add a few more temporal models

### PR DESCRIPTION
**Pull request by bkhelifi**
**Description**
This pull request associated to the issue #3325 adds 3 temporal models: LinearTemporalModel, PowerLawTemporalModel and SinusTemporalModel.

**Dear reviewer**
One might need to discuss about the exact functions...

### CODE REVIEW

**Comment by QRemy** [comment]
Thanks @bkhelifi for the implementation of these new models. Here are some suggestions:
- you could add their example pages in gammapy/examples/models/temporal/ so they will appear in the Gallery with the others (https://docs.gammapy.org/dev/modeling/gallery/index.html#temporal-models).
- you could add them in the test_io:: make_all_models : https://github.com/gammapy/gammapy/blob/0af7bc31efe423fdf95f3b70ec8332a0a081a658/gammapy/modeling/models/tests/test_io.py#L227
- tests for the serialisation using to_dict/from_dict are also missing for the temporal models (except for the LightCurveTemplateTemporalModel).

File: gammapy/modeling/models/__init__.py [hunk 1]
```diff
@@ -52,9 +52,12 @@
 TEMPORAL_MODEL_REGISTRY = Registry(
     [
         ConstantTemporalModel,
+        LinearTemporalModel,
         LightCurveTemplateTemporalModel,
         ExpDecayTemporalModel,
         GaussianTemporalModel,
+        PowerLawTemporalModel,
+        SinusTemporalModel,
```
Comment (registerrier) [inline gammapy/modeling/models/__init__.py:60]: Maybe `SineTemporalModel` or `SinusoidalTemporalModel`?

Comment (bkhelifi) [reply gammapy/modeling/models/__init__.py:60]: fixed

File: gammapy/modeling/models/temporal.py [hunk 2]
```diff
@@ -170,18 +170,70 @@ def integral(self, t_min, t_max):
         return (t_max - t_min) / self.time_sum(t_min, t_max)
 
 
+class LinearTemporalModel(TemporalModel):
+    r"""Temporal model with a linear variation.
+
```
Comment (adonath) [inline gammapy/modeling/models/temporal.py:175]: Add a forward reference to the example plot here, like `For more information see :ref:`linear-temporal-model`.`

File: gammapy/modeling/models/temporal.py [hunk 3]
```diff
@@ -418,5 +470,112 @@ def from_dict(cls, data):
...
+        """Create dict for YAML serialisation"""
         return {"type": self.tag[0], "filename": self.filename}
+
+
+class PowerLawTemporalModel(TemporalModel):
+    r"""Temporal model with a Power Law decay.
+
+    .. math::
+            F(t) = ((t - t_ref)/t0)^alpha
+
```
Comment (adonath) [inline gammapy/modeling/models/temporal.py:482]: Add forward reference to example plot, like `For more information see :ref:`powerlaw-temporal-model`.`

File: gammapy/modeling/models/temporal.py [hunk 4]
```diff
@@ -418,5 +470,112 @@ def from_dict(cls, data):
...
+            value = np.log((t_max-t_ref)/(t_min-t_ref))
+            return t0 * value / self.time_sum(t_min, t_max)
+
+
+class SineTemporalModel(TemporalModel):
+    r"""Temporal model with a sinusoidal modulation.
+
+    .. math::
+            F(t) = 1 + amp * sin(omega*(t-t_ref))
+
```
Comment (adonath) [inline gammapy/modeling/models/temporal.py:537]: Add forward reference as in the other examples...

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks a lot @bkhelifi, I have left some minor inline comment otherwise looks good to me.

File: gammapy/modeling/models/temporal.py (hunk 4 above)
Comment (bkhelifi) [reply gammapy/modeling/models/temporal.py:534]: added for the 3

**Review by adonath — APPROVED** [review]
Thanks a lot @bkhelifi, no further comments from my side...

**Comment by adonath** [comment]
@bkhelifi Can you do one more follow up PR adding units and panel to the time axes in the example plots: https://docs.gammapy.org/dev/modeling/gallery/index.html#temporal-models

Thanks!

**Pull request merged**